client/app/components/form/widgets/MultiSelectWidget.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+
+import { Autocomplete, Chip, MenuItem, TextField } from "@mui/material";
+import { WidgetProps } from "@rjsf/utils/lib/types";
+import { DARK_GREY_BG_COLOR, BC_GOV_SEMANTICS_RED } from "@/app/styles/colors";
+
+interface Option {
+  const: string | number;
+  title: string | number;
+}
+
+const MultiSelectWidget: React.FC<WidgetProps> = ({
+  id,
+  onChange,
+  rawErrors,
+  schema,
+  value,
+  uiSchema,
+}) => {
+  const fieldSchema = schema.items as {
+    enum: Array<string | number>;
+    enumNames?: Array<string | number>;
+    type: string;
+  };
+
+  // Using enum and enumNames as anyOf was triggering a lot of validation errors for multiselect
+  // If no enumNames are provided, use the enum values as the names. The enumNames will not be saved in the formData.
+  const enumValues = fieldSchema?.enum;
+  const enumNames = fieldSchema?.enumNames;
+  const options = enumValues.map((val: string | number, index: number) => ({
+    const: val,
+    title: enumNames?.[index] || val,
+  }));
+
+  const handleChange = (e: React.ChangeEvent<{}>, option: Array<Option>) => {
+    onChange(option.map((o: Option) => o.const));
+  };
+
+  const placeholder = uiSchema?.["ui:placeholder"]
+    ? `${uiSchema["ui:placeholder"]}...`
+    : "";
+
+  const displayPlaceholder = value?.length === 0;
```

Comment:
I'm surprised the mui component doesn't have this check built-in

Reply:
No by default the placeholder stays. I made the decision to conditionally hide it once the user has selected something as depending on the chip length it would hide some of it as shown here:

<img width="968" alt="Screenshot 2023-12-05 at 3 23 43 PM" src="https://github.com/bcgov/cas-registration/assets/14259474/e946392a-378d-41b6-bd73-71bf009d2da4">


However I did find an example of always putting the placeholder on a new line. There is a demo here though this might be a question for the designers if they want further refinement.
https://codesandbox.io/p/sandbox/67090943how-can-i-set-the-placeholder-of-autocomplete-in-a-new-line-u8vbz